aten/src/ATen/native/native_functions.yaml

```diff
@@ -6475,6 +6475,36 @@
     SparseCPU, SparseCUDA: norm_sparse
   autogen: native_norm.ScalarOpt_dim_dtype_out
 
+# TODO: add these to core aten opset
+# return values are: output, save_mean, save_var, reserve
+# reserve tensor is only used in cudnn
+# note: for cudnn and miopen backends, input, weight, bias, running_mean, and running_var are all expected to be contiguous
+- func: batch_norm_with_update(Tensor input, Tensor? weight, Tensor? bias, Tensor(a!) running_mean, Tensor(b!) running_var, float momentum, float eps) -> (Tensor, Tensor, Tensor, Tensor)
```

Comment:
Do we want these on torch. ?
If so we should properly document them, otherwise, we should make them private with a leading _*

Reply:
Ok. I think we decided not to expose them in the torch. namespace at least initially, since these are only meant to be called from `aten.batch_norm`. I can add the leading underscore